logrotate.c

```diff
@@ -2121,6 +2137,51 @@ static int prerotateSingleLog(const struct logInfo *log, unsigned logNum,
     return hasErrors;
 }
 
+static void resumeSingleLog(struct logNames *rotNames, struct resumeConfigs *resCfgs)
+{
+    int i;
+    char *newName = NULL;
+    char *oldName;
+    const char *fileext = resCfgs->fileext;
+    const char *compext = resCfgs->compext;
+    int rotateCount = resCfgs->rotateCount;
+    int logStart = resCfgs->logStart;
+
+    if (asprintf(&oldName, "%s/%s.%d%s%s", rotNames->dirName,
```

Comment:
The code that constructs `oldname` should be moved to a helper function so that it is maintained at a single place.

Reply:
This place can indeed be refined. we can optimize it next time because it is also used elsewhere